keps/sig-node/2400-node-swap/README.md

```diff
@@ -166,6 +172,83 @@ administrators can configure the kubelet such that:
+
+By doing so, we can ensure a thorough understanding of the feature's performance and stability before considering the manual input of swap values in a subsequent beta release. This cautious approach will ensure the efficient allocation of resources and the smooth integration of swap support into Kubernetes.
+
+Allocate the swap limit equal to the requested memory for each container and adjust the proportion of swap based on the total swap memory available.
+
+#### Set Aside Swap for System Critical Daemon
+
+System critical daemons (such as Kubelet) are essential for node health. Usually, an appropriate portion of system resources (e.g., memory, CPU) is reserved as system reserved. However, swap doesn't inherently support reserving a portion out of the total available. For instance, in the case of memory, we set `memory.min` on the node-level cgroup to ensure an adequate amount of memory is set aside, away from the pods, and for system critical daemons. But there is no equivalent for swap; i.e., no `memory.swap.min` is supported in the kernel. 
+
+Since this proposal advocates enabling swap only for the Burstable QoS pods, this can be done by setting `memory.swap.max` on the cgroups used by the Burstable QoS pods. The value of this `memory.swap.max` can be calculated by:
+
+memory.swap.max = total swap memory available on the system - system reserve (memory)
+
+This is the total amount of swap available for all the Burstable QoS pods; let's call it `TotalPodsSwapAvailable`. This will ensure that the system critical daemons will have access to the swap at least equal to the system reserved memory. This will indirectly act as having support for swap in system reserved.
+
+#### Use Swap Throttling
+
+`memory.swap.high` in the cgroup v2 allows you to enable throttling for swap allocation. If a cgroup’s swap usage exceeds `memory.swap.high`, all its further allocations will be throttled.
+
+Throttling will be enabled at individual Burstable QoS pod as well as at Burstable QoS cgroup level.
+
+1. At the Burstable QoS cgroup level, it can be calculated with: memory.swap.high = `TotalPodsSwapAvailable` * swap throttling factor
+2. At individual Burstable QoS pod level, it can be calculated with: memory.swap.high = memory.swap.max * swap throttling factor
+
+The default value of the memory throttling factor is set to 0.9.
+### Steps to Calculate Swap Limit
+
+1. **Calculate the total memory requests of the pod:**
+   - Sum up the memory requests of all containers in the pod. Let's call this value `TotalMemory`.
+
+2. **Determine the swap proportion for each container:**
```

Comment:
It may be desired by the end user to NOT have swap for container which has `request== limit` (for a specific container, even in a burstable Pod). So maybe this step can be adjusted and those containers which has `request==limit` are excluded from the swap proportion calculations.

This will also be good from security perspective. If you want to disable swap for your Pod - just make it's limit equal to request.

Reply:
Sounds good to me :+1: